### ISSUE TITLE
Feature request: alignment scoring scale with splint length

### ISSUE THREAD

**Issue by gordonrix** [christopher-vollmers/C3POa#1]
I am implementing this pipeline in a pipeline I am building. I am using golden gate instead of gibson to generate the circular product, so my 'splints' are much smaller than the example splints. This led to a silent failure mode in which most sequences were parsed and everything looked normal, but there were no sequences in the output files. An error describing what the problem was would have been helpful, but I was able to find that changing the scoring settings in `C3POa.py` did the trick:

`penalty, iters, window, order = 40, 3, 9, 2`

I require automated installation of all tools in my pipeline, and making this change automatically can be done but would be a bit hacky. Would be much better if this could be an argument or even better if the scoring was tuned for the length of the splint (and probably also the fraction of ambiguous bases in the splint)

Also, I believe the dependency editdistance is not installed by either the setup.sh script or the conda environment .yaml

Thanks for making this tool!

**Comment by christopher-vollmers** [comment]
Hi Gordon,

Thank you for pointing out that I was missing editdistance in setup script and conda environment. I fixed that now. 
As for your actual request: Let me look into it. We originally dialed in those parameters entirely empirically. I wouldn't know how we could adjust them automatically based on splint length. Or at least not without a lot of work. But you are not the first person to contact me about shorter splints. I think making those parameters a single argument should be fairly straightforward. 
I'll get back to you on that!

Best,
Chris

**Comment by gordonrix** [comment]
Awesome, thanks so much!

**Comment by christopher-vollmers** [comment]
Would you mind testing an executable for me? I think I implemented custom peak finder settings but don't have short splint data to test it

**Comment by gordonrix** [comment]
Of course! and thanks for working on this! I managed to get the automated
install within a conda environment to work using a search and replace
command for that line, but this would be much better. I'm also attaching a
.fastq file containing 25 RCA reads, most of which should be 3x+
concatemers. The short splint sequence for these is:

CTCGAGGcgaatttcttatgNNNYRNNNYRNNNYRNNNCACTCGCACTGACTCGATNNNNNNNCGTCAGGACTGCNNNNNNNCGTGTAGAGACTGCGTAGGNNNYRNNNYRNNNYRNNNctaaggagtacaccaaaacac

If it's easier for me to test it, I'm happy to of course.

**Comment by christopher-vollmers** [comment]
What would be the best way to send you the executable? I can't attach it here apparently.

**Comment by gordonrix** [comment]
put it on a branch in the C3POa GitHub perhaps?

**Comment by christopher-vollmers** [comment]
Okay. Still learning this stuff. I made a branch and pushed the new executable. Try it to see whether it makes sense

**Comment by gordonrix** [comment]
hah! Me too! Just made my first branch the other day. Should be able to
find the time to check this out this week, thanks!

**Comment by gordonrix** [comment]
This took longer than expected to get to, as with all things, but it looks
like it's working for me, thanks!

**Comment by gordonrix** [comment]
Would you be able to merge this branch with the most recent versions? The feature works well for me, but that branch is no longer compatible with newer versions of numpy

**Comment by mcrone** [comment]
Just to expand on this, is it possible to use this pipeline when there is no splint sequence? I am planning to use RCA for plasmid sequencing and then create the consensus using this tool. In this case there would be no specific splint.

**Comment by christopher-vollmers** [comment]
I recommend Tidehunter for that purpose. Having a splint sequence is a core requirement for C3POa and there is really no way around it.

Best,
Chris

**Comment by mcrone** [comment]
Thank you for the suggestion! I’d just come across that about an hour after posting.
